cmd/crank/beta/validate/cmd.go

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright 2024 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package validate implements schema validation of Crossplane resources.
+package validate
+
+import (
+	"github.com/alecthomas/kong"
+
+	"github.com/crossplane/crossplane-runtime/pkg/errors"
+	"github.com/crossplane/crossplane-runtime/pkg/logging"
+)
+
+// Cmd arguments and flags for render subcommand.
+type Cmd struct {
+	// Arguments.
+	Schemas   string `arg:"" help:"A YAML file with XRD and CRD schemas."`
```

Comment:
Cross-ref to https://github.com/crossplane/crossplane/issues/5162#issuecomment-1878824708.

Supporting XRDs and CRDs "directly" as an option makes sense to me, but I'd like to support packages too. For example:

```shell
# Validate that the rendered resources are valid EKS v0.40.0 resources.
crossplane beta render ... | crossplane beta validate  --provider=upbound/provider-aws-eks:v0.40.0
```

As I mentioned in the linked comment, I'd like to move us further toward a world where CRDs are an implementation detail. Not something folks need to think about, unless they want to.

Reply:
Also, is this a file or a directory? Or could it be either? The help says "a YAML file" but the examples imply a directory.

Reply:
+1 for @negz idea here on supporting higher level crossplane package concepts instead of just raw schemas dir 👍

Reply:
I agree with supporting crossplane packages like providers or configurations. However, to support those, we need to download package images with their dependencies and extract CRDs/XRDs. Since, it's not expected to have image version changes frequently, it makes sense to implement a caching mechanism as well. These two will take some time to implement. Therefore, I suggest to have this PR as an initial version and iterate on it.